modules/ztp-creating-siteconfig-custom-resources.adoc

```diff
@@ -0,0 +1,343 @@
+// Module included in the following assemblies:
+//
+// *scalability_and_performance/ztp-zero-touch-provisioning.adoc
+
+[id="ztp-creating-siteconfig-custom-resources_{context}"]
+= Creating SiteConfig custom resources
+
+A SiteConfig custom resource (CR) contains the site-specific data required to install and configure a cluster for RAN applications. The ZTP process uses the data in this CR to generate additional CRs necessary for installation using the Assisted Service operator and day-2 configuration of the cluster.
+
+For each SiteConfig in `<site>.yaml`, the data is used to create installation custom resources (CRs) in a directory named for the cluster.
+
+For every site a directory is created with the site name and for every cluster a file with the cluster name is created. The cluster file contains the CRs in the table. Every cluster has its own namespace and all the CRs are under that namespace. The namespace and the CR names are the same as the cluster name. Secret and ClusterImageSet are not generated by SiteConfig. You have to create them separately due to security concerns.
+
+The following table describes the custom resources, the generated filenames, and usage in ZTP.
+
+[cols="1,1,1"]
+|===
+| Resource (filename) | Description | Usage
+
+|BareMetalHost
+(`BareMetalHost.yaml`)
+|Contains the connection information for the Baseboard Management controller (BMC) of the target bare metal machine.
+|Provides access to the BMC in order to load and boot the Discovery ISO on the target machine using the Redfish protocol.
+
+|InfraEnv
+(`InfraEnv.yaml`)
+|Contains information for pulling OCP onto the target bare metal machine.
+|Used with ClusterDeployment to generate the Discovery ISO for the SNO cluster.
+
+|AgentClusterInstall
+(`AgentClusterInstall.yaml`)
+|Specifies the SNO cluster’s configuration such as networking and the number of supervisor (control plane) nodes. Shows the kubeconfig and credentials once the installation is complete.
+|Specifies the SNO cluster configuration information and provides status during the installation of the cluster.
+
+|ClusterDeployment
+(`ClusterDeployment.yaml`)
+|References the AgentClusterInstall to use.
+|Used with InfraEnv to generate the Discovery ISO for the SNO cluster.
+
+|NMStateConfig
+(`NMStateConfig.yaml`)
+|Provides network configuration information such as `MAC` ->`IP` mapping, DNS server, default route, and other network settings. Not needed if DHCP is used.
+|Sets up static IP for the SNO cluster’s Kube API server.
+
+|Agent
+|Contains hardware information about the target bare metal machine. This is created automatically during AI post installation steps.
+|Created automatically on the hub when the target machine's Discovery ISO boots.
+
+|ManagedCluster
+(`ManagedCluster.yaml`)
+|When a cluster will be managed by the hub, it needs to be imported and known. This Kubernetes object provides that interface.
```

Comment:
Avoid future tense "When a cluster is managed by the hub..."

Reply:
changed.